src/main/java/org/opensearch/security/dlic/rest/api/InternalUsersApiAction.java

```diff
@@ -144,8 +148,28 @@ protected void handlePut(RestChannel channel, final RestRequest request, final C
             throw new IOException(ex);
         }
 
+        // for existing users, hash is optional
+        if (userExisted && securityJsonNode.get("hash").asString() == null) {
```

Comment:
For my understanding, why don't we need hash for existing users?

Reply:
You don't need the hash for existing users since it is already stored in our records. So as long as we have it we are fine with them not providing it.